java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/reflectionmodel/ReflectionTypeParameter.java

```diff
@@ -76,7 +76,7 @@ public boolean equals(Object o) {
     @Override
     public int hashCode() {
         int result = typeVariable.hashCode();
-        result = 31 * result + (declaredOnClass ? 1 : 0);
+        result = 31 * result + (declaredOnType() ? 1 : 0);
```

Comment:
That sounds reasonable. However, I'm quite unfamiliar with Java HashCodes so I have to ask: Are you thinking of just adding the hashcode of the container to the `result * 31`?